src/Parser/InputParser/InputParser.cpp

```diff
@@ -0,0 +1,170 @@
+/*
+** EPITECH PROJECT, 2022
+** plaza
+** File description:
+** InputParser
+*/
+
+#include "InputParser.hpp"
+#include "Exception.hpp"
+#include <iostream>
+
+InputParser::InputParser()
+{
+    ParserEX("Initialisation of the InputParser class", Logger::INFO);
+    _command = "";
+    _argumentNumber = 0;
+    _separator = ';';
+    _arguments.clear();
+}
+
+void InputParser::setCommand(const std::string &command)
+{
+    std::string tmpCommand = command;
+
+    ParserEX("Set the command for the InputParser class", Logger::INFO);
+    _command = command;
+    while (!tmpCommand.empty()) {
+        while (tmpCommand.find(_separator) == 0)
+            tmpCommand = tmpCommand.substr(1);
+        if (tmpCommand.empty())
+            break;
+        _arguments.push_back(tmpCommand.substr(0, tmpCommand.find(_separator)));
+        if (tmpCommand.find(_separator) < tmpCommand.size())
+            tmpCommand = tmpCommand.substr(tmpCommand.find(_separator));
+        else
+            tmpCommand.clear();
+        _argumentNumber++;
+    }
+}
+
+const std::string &InputParser::getCommand() const
+{
+    ParserEX("Get the command for the InputParser class", Logger::INFO);
+    return (_command);
+}
+
+void InputParser::prossessArguments()
+{
+    std::vector<std::string> tmp = _arguments;
+    std::vector<std::string> args;
+
+    ParserEX("Prossess the arguments for the InputParser class", Logger::INFO);
+    setCommandSeparator(' ');
+    for (auto it: tmp) {
+        _arguments.clear();
+        setCommand(it);
+        for (auto arg: _arguments) {
+            args.push_back(arg);
+        }
+        args.push_back(";");
+    }
+    _arguments = args;
+    _checkArgumentsValidity();
+}
+
+bool InputParser::_isValidePizzaName(const std::string &pizza) const
+{
+    const std::string availablePizza[4] = {
+            "Regina",
+            "Margarita",
+            "Americana",
+            "Fantasia"};
+    bool findValidePizza = false;
+
+    for (std::size_t index = 0; index < 4; index++) {
+        if (availablePizza[index] == pizza)
+            findValidePizza = true;
+    }
+    if (!findValidePizza)
+        ParserEX("Invalide pizza name in the command", Logger::LOW);
```

Comment:
Already better, Invalid w/o an e would be ideal
Consider adding the faulty pizza name to `what()`